sound/soc/sof/intel/hda.c

```diff
@@ -328,27 +605,44 @@ int hda_dsp_probe(struct snd_sof_dev *sdev)
 	sdev->hda = hdev;
 	hdev->desc = chip;
 
-	/* HDA base */
-	sdev->bar[HDA_DSP_HDA_BAR] = pci_ioremap_bar(pci, HDA_DSP_HDA_BAR);
-	if (!sdev->bar[HDA_DSP_HDA_BAR]) {
-		dev_err(&pci->dev, "error: ioremap error\n");
-		/*
-		 * FIXME: why do we return directly,
-		 *  should we have a goto err here?
-		 *  or should all these gotos be replaced
-		 * by a return?
-		 */
-		return -ENXIO;
-	}
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
+	hbus = devm_kzalloc(&pci->dev, sizeof(*hbus), GFP_KERNEL);
+	if (!hbus)
+		return -ENOMEM;
+
+	sdev->hbus = hbus;
+	bus = &hbus->core;
+
+	/* HDA bus init */
+#if IS_ENABLED(CONFIG_SND_SOC_HDAC_HDA)
+	ext_ops = snd_soc_hdac_hda_get_ops();
+#endif
+	snd_hdac_ext_bus_init(bus, &pci->dev, NULL, NULL, ext_ops);
```

Comment:
the defines are probably wrong, if IS_ENABLED(CONFIG_SND_SOC_HDAC_HDA) is false, the ext_ops is not initialized. Is this is intentional you've got to comment it.

Reply:
um, this comes from cAVS, no chance to debug it yet, if soc_hdac_hda unselected, I think it may pass ext_ops with NULL and no ops(e.g. attach/detach) will be processed?